### ISSUE TITLE
Assistance Request - Specific adaptation of Microsoft.Identity.Web

### ISSUE THREAD

**Issue by sujayvsarma** [Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2#250]
This actually follows from [this issue](https://github.com/dotnet/aspnetcore/issues/18117). 

My scenario is that I have an application that has to do with managing Azure-based resources (you know, VMs, apps, etc). The application is multi-tenanted -- anyone that can logon to Azure Portal and use that should be able to login to this application and use it. Currently I have a wide-net, meaning Multi-tenant + MS account, because I am not sure if the application can get away with being a pure Multi-tenant not requiring the "+ MS account" part.  

Anyway, so I am already using `Microsoft.AspNetCore.Authentication.AzureAD.UI` as a NuGet. I don't need the B2C part. 

Now, after reading through all the code in `Microsoft.Identity.Web` and the `WebApp-OpenIDConnect-DotNet` project in `2-3-Multitenant` from this repo, I have the following in mind:

1. I see that the primary USP of `Microsoft.Identity.Web` is that it reconfigures the underlying `Microsoft.AspNetCore.Authentication.AzureAD.UI` library and does some of the activities slightly differently than that upstream package -- for example, it adds the ability to run off the `common` endpoint and turns off issuer validation. 

2. Another thing I liked about this, especially as implemented n the 2-3 sample is the ability to check the Azure Tenant against an "onboarded" list and then go through the Admin Consent process, which is something I definitely need.

SO... 

Let's say I don't want the whole of Microsoft.Identity.Web sitting in my code base (& frankly, some 60% of it seems to be either unapplicable to me, or not used anywhere at all), how do I copy/paste bits of it into my application so that it does all of this? What are the parts that are really important to do?

**Comment by jmprieur** [comment]
@sujayvsarma what if it was available as a NuGet package? In the end this would be a component, and you'd use the public API you'd need. Did you see https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/master/Microsoft.Identity.Web/README.md ?

**Comment by jmprieur** [comment]
Also, today, Microsoft.Identity.Web leverages `Microsoft.AspNetCore.Authentication.AzureAD.UI` but we are experimenting about leveraging the OIDC middleware directly. See the [removingUis](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/commits/jmprieur/removingUis) branch

**Comment by sujayvsarma** [comment]
Yes, exactly. I have responded to your survey [with my comments](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/issues/69#issuecomment-571912449). 

But this question has nothing to do with that. This is on how I would "minify" the current Identity.Web codebase for specific usage in my application rather than having to fork/clone and use the whole thing with all of its other dependencies that I do not want -- eg: EntityFramework.

**Comment by sujayvsarma** [comment]
> Also, today, Microsoft.Identity.Web leverages Microsoft.AspNetCore.Authentication.AzureAD.UI but we are experimenting about leveraging the OIDC middleware directly. See the removingUis branch

So basically you would be adding to the confusion of profusion in the packages that do the same thing! Please dont!

**Comment by sujayvsarma** [comment]
Again.. this issue is not a bug report or a request for changing Microsoft.Identity.Web.

This is a question, asking for your help in adapting that code for my needs.

**Comment by jmprieur** [comment]
@sujayvsarma : Do you mean you'd want to have more NuGet packages doing less things?

**Comment by sujayvsarma** [comment]
> you mean you'd want to have more NuGet packages doing less things?

Yes. I would prefer that.

**Comment by jmprieur** [comment]
> > Also, today, Microsoft.Identity.Web leverages Microsoft.AspNetCore.Authentication.AzureAD.UI but we are experimenting about leveraging the OIDC middleware directly. See the removingUis branch
> 
> So basically you would be adding to the confusion of profusion in the packages that do the same thing! Please dont!

Actually we are discussing with the ASP.NET Core team about that. It's not something we'd do on our own. cc: @blowdart

**Comment by multco-malan** [comment]
Hi Sujay,
What does USP mean?
thanks

**Comment by sujayvsarma** [comment]
@multco-malan - USP: Unique Selling Points

What's unique about this? Why should people care it exists? etc

**Comment by kalyankrishna1** [comment]
You are on a path that will consistently make your application outdated. The Microsoft.Identity.Web is a library, that will continue to evolve for a long time. Your application is probably using 1%- 5% of the libraries that it has referenced, so how many references would you continue to minify for your project ?

While no one has stopped you from downloading the library  and minify it yourself, the onus would be on you to keep your copy updated as the library itself is enhanced, get bugs fixes and such

**Comment by jmprieur** [comment]
Also, when you deploy your application, depending on the host, the parts of Microsoft.Identity.Web that are not used (see technics like .NET native)

For the unique selling points, @sujayvsarma 

![image](https://user-images.githubusercontent.com/13203188/88519636-b392d080-cff2-11ea-978b-712caf1d1fe3.png)

**Comment by sujayvsarma** [comment]
2 things, given this was a very old ask that someone misguidedly closed in the middle of a discussion! -

1] I had since succeeded in minifying the library for my use. It was very very tiny. I also ended up having to rewrite small bits of it and add more handlers. Still the code was smaller than the `Microsoft.Identity.Web` codebase.

This was the primary reason that I let this item remain closed.

2] I dont much care for web hosts. I always use an SCD [self contained deployment].

Finally, I have completely moved out of IT. So this is currently irrelevant to me.
